### PR TITLE
Add a flag to disable the ipcache check

### PR DESCRIPTION
**Pull request by michi-covalent**
Add `--skip-ip-cache-check` flag with the default set to true. This is
meant to be a temporary flag while we investigate what's causing the flake.

Ref: #361

Signed-off-by: Michi Mutsuzaki <michi@isovalent.com>

### CODE REVIEW

**Comment by michi-covalent** [comment]
aks cluster creation failed with this error. re-triggering.

https://github.com/cilium/cilium-cli/actions/runs/1184098825

> ERROR: (ReconcileStandardLoadBalancerError) AKS encountered an internal error while attempting the requested Creating operation. AKS will continuously retry the requested operation until successful or a retry timeout is hit. Check back to see if the operation requires resubmission. Correlation ID: cc4b16c3-e207-4ec0-935b-172e2abdbb8b, Operation ID: 597493aa-980d-41f3-8c4c-5a418e13f412, Timestamp: 2021-08-30T23:27:08Z.

**Comment by michi-covalent** [comment]
hitting the same issue again. not sure how to proceed 😞

File: internal/cli/cmd/connectivity.go
Comment (tklauser) [inline internal/cli/cmd/connectivity.go:122]: Should we mark this flag as hidden so we can easily remove it again? Given that we intend to fix the ipcache flake #361 eventually,

**Review by tklauser — APPROVED** [review]

**Review by bmcustodio — APPROVED** [review]

File: internal/cli/cmd/connectivity.go
Comment (bmcustodio) [reply internal/cli/cmd/connectivity.go:122]: Hmmm this makes sense, but at the same time I think that if users face the issue and the flag is hidden they won't know they can use it to proceed.

Comment (tklauser) [reply internal/cli/cmd/connectivity.go:122]: Given that we default the flag to `true` (i.e. skip the ipcache check by default), I think users should not face the ipcache issue anymore. They would need to figure out to run with `--skip-ip-cache-check=false` at which point they already figured out the flag :wink:

Comment (bmcustodio) [reply internal/cli/cmd/connectivity.go:122]: Oh right, makes total sense 💯 — I hadn't looked at the default.

**Review by twpayne — APPROVED** [review]

**Comment by nbusseneau** [comment]
> hitting the same issue again. not sure how to proceed disappointed

This is an infrastructure issue, waiting for AKS support atm: https://github.com/cilium/cilium/issues/17260

**Comment by michi-covalent** [comment]
re-triggered to see if aks happy now 👀

File: internal/cli/cmd/connectivity.go
Comment (tklauser) [reply internal/cli/cmd/connectivity.go:122]: Looks like this hasn't been addressed before merging. I've opened another PR: #508

Comment (michi-covalent) [reply internal/cli/cmd/connectivity.go:122]: thanks @tklauser , i just saw the green merge button and pushed it 😅

**Pull request merged**
